pkgs/tools/graphics/directx-shader-compiler/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, fetchFromGitHub, cmake, python3, git }:
+
+stdenv.mkDerivation rec {
+  pname = "directx-shader-compiler";
+  version = "1.5.2010";
+
+  # Put headers in dev, there are lot of them which aren't necessary for
+  # using the compiler binary.
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "Microsoft";
+    repo = "DirectXShaderCompiler";
+    rev = "b640fa4ba4afb8eb9f15b4a34ab5187c7d94de97";
+    sha256 = "0ccfy1bfp0cm0pq63ri4yl1sr3fdn1a526bsnakg4bl6z4fwrnnj";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake git python3 ];
+
+  configurePhase = ''
+    # Requires some additional flags to cmake from a file in the repo
+    additionalCMakeFlags=$(< utils/cmake-predefined-config-params)
+    cmakeFlags="$additionalCMakeFlags''${cmakeFlags:+ $cmakeFlags}"
+    cmakeConfigurePhase
+  '';
+
+  # The default install target installs heaps of LLVM stuff.
+  #
+  # Upstream issue: https://github.com/microsoft/DirectXShaderCompiler/issues/3276
+  #
+  # The following is based on the CI script:
+  # https://github.com/microsoft/DirectXShaderCompiler/blob/master/appveyor.yml#L63-L66
+  installPhase = ''
+    mkdir -p "$out/bin"
+    mv bin/dxc* "$out/bin/"
```

Comment:
Can we use install here?

Reply:
sure, but what does it offer over mv or cp?

Reply:
using `install` you don't need `mkdir`.

Reply:
and you can make sure the permission are correct.

Reply:
> using install you don't need mkdir.

but I've already written `mkdir`...

> and you can make sure the permission are correct

I should hope that the build system has already done that, writing them myself here would just be repeating that.

Besides, install has poor support for copying directories recursively (as used with the "include" directory)

I'll make the change if you both insist.

Reply:
Are there more than three binaries? If so I think mv is fine to not clutter to many installs.

Reply:
There is one binary and one symlink to it, but I don't want to write the names in the nix expression as it makes it all more brittle (especially since the binary includes the version number).